File: 프로그래머스/Level2/고득점 Kit/위장-해시.py
def solution(clothes):
    answer = 0
    d = dict()
    
    for cloth in clothes :
        if cloth[1] not in d.keys() :
            d[cloth[1]] = 1
        else :
            d[cloth[1]] += 1
        
    num = 1
    for k in d.keys() :
        num *= (d[k]+1)
    
    answer = num - 1
    
    return answer

File: Source Code/c++/프로그래머스/Level2/고득점 Kit/위장-해시.cpp
#include <string>
#include <vector>
#include <map>
#include <iostream>
using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    map<string, int> m;
    
    for(int i=0; i<clothes.size(); i++){
        m[clothes[i][1]]++;
    }
    
    map<string, int>::iterator it;
    
    for(it=m.begin(); it!=m.end(); it++){
        answer = answer * (it->second+1);
    }
    
    answer = answer - 1;
    
    return answer;
}